keps/sig-node/3066-subsecond-probes/README.md

```diff
@@ -0,0 +1,905 @@
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+The Probe struct contains int32 fields that specify seconds for timeouts.
+Some users would like to have timeouts less than one second.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+#### Istio
+
+TODO(???): explain this scenario, presumably around sidecar readiness?
+
+#### Knative
+
+Knative will create Pods (via Deployment) and wait for them to become `Ready`
+when handling an HTTP request from an end-user. In this case, Pod readiness
+latency has a direct impact on HTTP request latency. (In the steady state,
+Knative will re-use an existing Pod, but this situation can happen on scale-up,
+and is guaranteed to happen on scale-from-zero.)
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+An ability to specify timeouts that are less than one second.
+Add additional tests cases to the timeout test cases.
+
+? link to existing test cases ?
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+V2 API for existing objects.
+Converting fields from int32 to resource.Quantity.
+
+
+## Proposal
+
+Add a new struct, `ProbeOffset`, with two subfields: `unit` (type string) and value (type int32). 
+
+That struct would be used to create three new fields in a Probe: 
+- `periodSecondsOffset`
+- `initialDelaySecondsOffset`
+- `timeoutSecondsOffset`
+
+The offset would be used to adjust the timing intervals within probes, by adding the offset value to the probe value. For example,
+
+    ```golang
+    ...
+    periodSeconds: 1
+    periodSecondsOffset:
+        - unit: Milliseconds
+          value: -500
+    ```
+
+would translate to a `periodSecond` value of `500ms`. 
+
+`periodSeconds: 1` and `offset: 500ms` would translate to `1.5s`. 
+
+To have an effective `periodSeconds` of `1.5`, either of the following could work:
+* `periodSeconds: 1`, `offset: 500ms`
+* `periodSeconds: 2`, `offset: -500ms`
+
+Each offset would be restricted to an absolute value less than `1s`, and the offset would never allow the adjusted interval to be less than 0 (if it was less than 0, it would get failed at the validation stage).
```

Comment:
```suggestion
Each offset would be restricted to a value less than `1s`, and the offset would never allow the adjusted interval to be less than 0 (if it was less than 0, it would get failed at the validation stage).
```